src/app/components/hris/employees/employee-profile/accordions/accordion-career-summary/accordion-salary-details/accordion-salary-details.component.ts

```diff
@@ -0,0 +1,158 @@
+      this.isAdminUser = true;
+    }
+  }
+
+  initializeSalaryDetailsForm(salaryDetails: EmployeeSalary) {
+    if (salaryDetails != null) {
+      this.salaryDetailsForm = this.fb.group({
+        remuneration: [salaryDetails.remuneration , [Validators.required, Validators.pattern(/^[0-9]*$/)]],
+      });
+      this.getDate();
+    }
+    else {
+      this.salaryDetailsForm = this.fb.group({
+        remuneration: [ "", [Validators.required, Validators.pattern(/^[0-9]*$/)]],
+      });
+    }
+    this.salaryDetailsForm.disable();
+  }
+
+  getEmployeeSalaryDetails() {
+    this.employeeSalaryService.getEmployeeSalary(this.employeeProfile.id as number).subscribe({
+      next: data => {
+        this.employeeSalary = data;
+        this.initializeSalaryDetailsForm(this.employeeSalary);
+      },
+      error: (error) => {
+        this.snackBarService.showSnackbar("Error fetching salary details", "snack-error");
+      }
+    })
+  }
+
+  getDate() {
+    let updateDate = this.employeeSalary.salaryUpdateDate;
+    let day = new Date(updateDate as Date).getDate();
+    const monthNames = ["January", "February", "March", "April", "May", "June",
+      "July", "August", "September", "October", "November", "December"];
+    let month = monthNames[new Date(updateDate as Date).getMonth()];
+    let year = new Date(updateDate as Date).getFullYear();
+    this.message = day + " " + month + " " + " " + year;
+  }
+
+  saveEmployeeSalaryDetails() {
+    const salaryDetailsFormValue = this.salaryDetailsForm.value;
+    if (this.salaryDetailsForm.valid) {
+      this.editSalary = false;
+      if (this.employeeSalary) {
+        this.employeeSalaryDetailsDto = {
+          employeeId: this.employeeSalary.id,
+          id: this.employeeSalary.id,
+          salary: this.employeeSalary.salary,
+          minSalary: this.employeeSalary.minSalary,
+          maxSalary: this.employeeSalary.maxSalary,
+          remuneration: salaryDetailsFormValue.remuneration,
+          band: this.employeeSalary.band,
+          contribution: this.employeeSalary.contribution,
+          salaryUpdateDate: new Date()
+        }
+        this.employeeSalaryService.updateEmployeeSalary(this.employeeSalaryDetailsDto).subscribe({
+          next: () => {
+            this.snackBarService.showSnackbar("Salary details has been updated", "snack-success");
+            this.getDate();
+            this.editSalary = false;
+            this.salaryDetailsForm.disable();
+          },
+          error: (error) => {
+            this.snackBarService.showSnackbar(error, "snack-error");
+          }
+        })
+      } else {
+        this.employeeSalaryDetailsDto = {
+          employeeId: this.employeeProfile.id,
+          id: 0,
+          salary: 0,
+          minSalary: 0,
+          maxSalary: 0,
+          remuneration: salaryDetailsFormValue.remuneration,
+          band: 0,
+          contribution: "",
+          salaryUpdateDate: new Date()
+        }
```

Comment:
To better clean up the code, this creation of this dto can be moved to a method that populates this dto object